core/state_transition.go

```diff
@@ -363,14 +375,18 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 		// Increment the nonce for the next transaction
 		st.state.SetNonce(msg.From(), st.state.GetNonce(sender.Address())+1)
 		ret, st.gas, vmerr = st.evm.Call(sender, st.to(), st.data, st.gas, st.value)
-		if vmerr == nil && *msg.To() == GetStateConnectorContract(chainID, timestamp) && len(st.data) >= 36 && len(ret) == 32 {
-			if GetStateConnectorActivated(chainID, timestamp) &&
-				bytes.Equal(st.data[0:4], SubmitAttestationSelector(chainID, timestamp)) &&
-				binary.BigEndian.Uint64(ret[24:32]) > 0 {
-				err = st.FinalisePreviousRound(chainID, timestamp, st.data[4:36])
-				if err != nil {
-					log.Warn("Error finalising state connector round", "error", err)
-				}
+		if vmerr == nil &&
+			*msg.To() == stateConnectorContract(chainID, timestamp) &&
+			len(st.data) >= 36 &&
+			len(ret) == 32 &&
+			isStateConnectorActivated(chainID, timestamp) &&
+			// 4 first bytes corresponds to the function name
+			bytes.Equal(st.data[0:4], submitAttestationSelector(chainID, timestamp)) &&
+			// determine whether this is read-only call
+			binary.BigEndian.Uint64(ret[24:32]) > 0 {
+			err = st.stateConnector.finalisePreviousRound(chainID, timestamp, st.data[4:36])
+			if err != nil {
+				log.Warn("error finalising state connector round", "error", err)
```

Comment:
Isn't this effectively the same as

```go
log.Warn("error finalising state connector round error", err)
```

? If so, I'd rather reformat it like such

```suggestion
				log.Warn("error finalising state connector round:", err)
```

Reply:
there should be an even number of arguments provided, that's why it's `"error", err`

Reply:
There should be an even number of arguments provided, that's why it's `"error", err`

Reply:
Ah okay my bad, I thought this was the standard `log` package but it's actually a structured logger from ethereum. Makes sense then!